app/react/src/server/cra-config.ts

```diff
@@ -71,43 +71,53 @@ export function isReactScriptsInstalled(requiredVersion = '2.0.0') {
 }
 
 export const getRules = (extensions: string[]) => (rules: RuleSetRule[]) =>
-  rules.reduce((craRules: any, rule: any) => {
-    // If at least one extension satisfies the rule test, the rule is one
-    // we want to extract
-    if (rule.test && extensions.some(normalizeCondition(rule.test))) {
-      // If the base test is for extensions, return early
-      return craRules.concat(rule);
-    }
+  rules.reduce(
+    (craRules, rule) => {
+      // If at least one extension satisfies the rule test, the rule is one
+      // we want to extract
+      if (rule.test && extensions.some(normalizeCondition(rule.test))) {
+        // If the base test is for extensions, return early
+        return craRules.concat(rule);
+      }
 
-    // Get any rules contained in rule.oneOf
-    if (!rule.test && rule.oneOf) {
-      craRules.push(...getRules(extensions)(rule.oneOf));
-    }
+      // Get any rules contained in rule.oneOf
+      if (!rule.test && rule.oneOf) {
+        craRules.push(...getRules(extensions)(rule.oneOf));
+      }
 
-    // Get any rules contained in rule.rules
-    if (!rule.test && rule.rules) {
-      craRules.push(...getRules(extensions)(rule.rules));
-    }
+      // Get any rules contained in rule.rules
+      if (!rule.test && rule.rules) {
+        craRules.push(...getRules(extensions)(rule.rules));
+      }
 
-    return craRules;
-  }, []);
+      return craRules;
+    },
+    [] as RuleSetRule[]
+  );
 
 const getStyleRules = getRules(cssExtensions.concat(cssModuleExtensions));
 
 export const getTypeScriptRules = (webpackConfigRules: RuleSetRule[], configDir: string) => {
   const rules = getRules(typeScriptExtensions)(webpackConfigRules);
-  // We know CRA only has one rule targeting TS for now, which is the first rule.
-  const babelRule = rules[0];
-  // Resolves an issue where this config is parsed twice (#4903).
-  if (typeof babelRule.include !== 'string') return rules;
+
   // Adds support for using TypeScript in the `.storybook` (or config) folder.
-  return [
-    {
-      ...babelRule,
-      include: [babelRule.include, path.resolve(configDir)],
+  return rules.reduce(
+    (accRules, rule) => {
```

Comment:
I can see that you've changed this so that all rules now resolve the path and config path - have you considered the case where there is another rule defined requires a different "include"?

I don't think it's an issue in this case, but did you look through the existing includes?